apparmor.d/groups/hyprland/hyprland

```diff
@@ -0,0 +1,71 @@
+# apparmor.d - Full set of apparmor profiles
+# Copyright (C) 2024 odomingao
+# SPDX-License-Identifier: GPL-2.0-only
+
+abi <abi/3.0>,
+
+include <tunables/global>
+
+@{exec_path} = @{bin}/Hyprland
+profile hyprland @{exec_path} flags=(attach_disconnected) {
+  include <abstractions/base>
+  include <abstractions/desktop>
+  include <abstractions/graphics>
+  include <abstractions/wayland>
+
+  capability sys_ptrace,
+
+  ptrace read,
+
+  network netlink raw,
+
+  signal send,
+
+  @{exec_path} mr,
+
+  @{bin}/** rPUx,
+  @{user_bin_dirs}/** rPUx,
+  owner @{user_share_dirs}/hyprpm/** mr,
+
+  /usr/share/hyprland/{,*} r,
+  /usr/share/libinput/{,*} r,
+
+  owner @{user_cache_dirs}/hyprland/** w,
+  owner @{user_config_dirs}/hypr/hyprland.conf r,
+
+  @{run}/systemd/sessions/@{int} r,
+  @{run}/udev/data/+acpi:* r,
+  @{run}/udev/data/+dmi:id r,
+  @{run}/udev/data/+drm:card@{int}-* r,
+  @{run}/udev/data/+input:input@{int} r,
+  @{run}/udev/data/+pci:* r,
+  @{run}/udev/data/+platform:pcspkr r,
+  @{run}/udev/data/+sound:card@{int} r,
+  @{run}/udev/data/+usb:* r,
+  @{run}/udev/data/c13:@{int} r,
+  @{run}/udev/data/c189:@{int} r,
+  @{run}/udev/data/c226:@{int} r,
+  owner @{run}/user/@{uid}/gamescope-* rw,
+  owner @{run}/user/@{uid}/hypr/{,**} rw,
+  owner @{run}/user/@{uid}/.hyprpaper_* rw,
+  owner @{run}/user/@{uid}/.hyprpicker_* rw,
+  owner /tmp/.X@{int}-lock w,
+  owner /dev/shm/@{uuid} rw,
```

Comment:
Move this to the wayland complete file.

Reply:
is it ok to remove `owner` from  `owner /dev/shm/@{uuid} rw,`? Applications running as a different user will not launch otherwise

Reply:
I would say no, (simply add a rule without owner in you local addition). This kind of runtime specific settings does not fit well with generic apparmor profiles. This kind would better be handled directly by the program that handle starting the app as a different user.